prboom2/src/g_game.c

```diff
@@ -3179,13 +3182,74 @@ void G_BeginRecording (void)
 {
   int i;
   byte *demostart, *demo_p;
+  int num_extensions = 0;
   demostart = demo_p = malloc(1000);
   longtics = 0;
 
-  if (umapinfo_loaded)
+  // ano - jun2019 - add the extension format if needed
+  if (gamemapinfo)
```

Comment:
Why is this changed from `umapinfo_loaded` to `gamemapinfo` here? AFAIUI, `gamemapinfo` is only set if there is UMAPINFO information available _for the current map_, whereas `umapinfo_loaded` is set a soon as a `UMAPINFO` lump is found and loaded. But what if we start a demo from a map without UMAPINFO changes and then migrate to UMAPINFO affected maps during the course of recording?

Reply:
Good catch -- it indeed oughta be `umapinfo_loaded`. I whipped up a test wad which contained UMAPINFO for maps 2, 3, and 4, recorded a demo starting from MAP01, and it skipped writing the extension header entirely. Got a fix incoming.